beta-src/src/utils/state/gameApiSlice/reducers/processUnitClick.ts

```diff
@@ -40,8 +40,10 @@ export default function processUnitClick(state, clickData) {
     }
   }
   // Destroy Units
+  const isDestroy =
+    currentOrders?.length && currentOrders[0].type === "Destroy";
```

Comment:
would recommend find or some here if were just checking for 1 item that meets the condition
`currentOrders.some((order) => order.type === "Destroy")`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some